### PR TITLE
1052-form-health-http

### LINKED ISSUE #981: Create New Health modal ui is not user friendly.
- [ ] improve the styling so that it looks like other modal's (ex: create incident)
- [ ] adjust field heights accordingly.

### LINKED ISSUE #1052: Form Health - Http

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #1052
Closes #981

### CODE REVIEW

File: src/components/Forms/Formik/FormikTemplateFields.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (!v) {
+          setFieldValue(name, undefined);
+        }
+      }}
+    >
+      <div className="flex flex-col p-4 space-y-2 border border-gray-200 rounded-md">
+        <FormikTextInput name={`${name}.template`} label="Template" />
+        <FormikTextInput name={`${name}.jsonPath`} label="JSON Path" />
+        <FormikCheckbox name={`${name}.expr`} label="Expr" />
+        <FormikCheckbox name={`${name}.javascript`} label="Javascript" />
```
Comment (moshloop) [inline src/components/Forms/Formik/FormikTemplateFields.tsx:53]: This should be a button group with Go Template | Javascript | Expr | JSONPath button, and then a editor with those languages use for autocomplete / highlighting

File: src/components/Forms/Health/HTTPHealthFormEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      <FormikJSONCheckFields
+        label="Response JSON Checks"
+        name={`${fieldName}.responseJSONChecks`}
+      />
+      <FormikTextInput
+        name={`${fieldName}.maxSSLExpiry`}
+        label="Max SSL Expiry"
+      />
+      <FormikTextInput name={`${fieldName}.method`} label="Method" />
+      <FormikTextInput name={`${fieldName}.ntlm`} label="ntlm" />
```
Comment (moshloop) [inline src/components/Forms/Health/HTTPHealthFormEditor.tsx:52]: Use a button group:
Authentication:  None | Basic | NTLM | NTLMv2
and then have username/password below which gating them - If you specify a auth method, you must specify username and password

File: src/components/Forms/Health/HTTPHealthFormEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+export function HTTPHealthFormEditor({ fieldName }: HTTPHealthFormEditorProps) {
+  return (
+    <>
+      <FormikTextInput name={`${fieldName}.name`} label="Name" required />
+      <FormikTextInput name={`${fieldName}.description`} label="Description" />
+      <FormikTextInput name={`${fieldName}.icon`} label="Icon" />
+      {/* add labels here */}
+      <FormikTemplateFields name={`${fieldName}.test`} label="Test" />
```
Comment (moshloop) [inline src/components/Forms/Health/HTTPHealthFormEditor.tsx:20]: Create form groups /  sections:

Name / Icon.  (one line)
Description
URL (alias to endpoint)
Method: GET (default) | PUT | POST |  HEAD (button group
Body: (If method ==  PUT/POST), textarea
Authentication:
Headers:

**Test** 
`HTTP Response Code`: (Use multi react-select component (Call it MultiSelectArrayField or similar) with a list of : 200,201,202,204,301,302,401,404,500,502,503) + the option to type in a new  value
`Max Response Time (in millis)`:
`Response Text`: 
*Exact response content expected to be returned by the endpoint*
`Max SSL Expiry Age (days)`: 
*Maximum number of days until the SSL Certificate expires.*
Script: `None | Go Template | Javascript | JSONPath | Expr` (when select a 3 line editor is shown below)

**Advanced**
`Customize Response`:  `None | Go Template | Javascript | JSONPath | Expr`  (display field)
`Transform Response`:  `None | Go Template | Javascript | JSONPath | Expr`  (display field)

**Comment by moshloop** [comment]
@mainawycliffe  

- Can we make the dialog title: `Add Health Check`
- Error when saving are not shown
- Can we  use a static value of "default" for namespace and not show it on the form
- Remove the back button if you choose the wrong type, you need to close the form and click add again
- Move the save button in to a fixed footer like all the other dialogs + remove save icon
- We will need a custom option when adding, that lets you put in the raw spec field.

**Comment by moshloop** [comment]
@mainawycliffe Its looking good - 

- [ ] The checkbox group button should shrink to fit 
- [ ] For the scripts we need a None item in addition to the other 4, that hides the script
- [ ] Remove the border around the template fields 
- [ ] The font size/color is not consistent for all the labels
- [ ] Can we make Advanced an "accordion" section collapsed by default
- [ ] The generated code needs to make the openapi schema/docs it has 
```yaml
advanced:
  customizeResponse:
```

**Comment by mainawycliffe** [comment]
@moshloop is the JSON Schema mapping correct now?

**Comment by moshloop** [comment]
@mainawycliffe - please output against https://docs.flanksource.com/canary-checker/reference/http/ and https://github.com/flanksource/canary-checker/raw/master/config/schemas/canary.schema.json

- username and password need to come under `authentication` - there is no authenticationMethod field, ntlm and ntlm2 are booleans

- maxSSLExpiry and thresholdMillis are numbers not text
- Missing schedule field and headers.
-

**Pull request merged**